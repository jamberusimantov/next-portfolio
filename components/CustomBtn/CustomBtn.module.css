.button {
    padding: .5rem 1rem;
    border: 1px solid var(--blue);
    background-color: var(--white);
    color: var(--blue);
    cursor: pointer;
    display: inline-flex;
    justify-content: center;
    align-items: center;
}

.button p {
    font-size: large;
    font-weight: 100;
    margin: 0;
}

.button:hover {
    background-color: var(--blue);
    color: var(--white);
}