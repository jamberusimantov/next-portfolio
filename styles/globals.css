html,
body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
    color: inherit;
    text-decoration: none;
}

* {
    box-sizing: border-box;
}

.hamburger {
    width: 40px;
    height: 40px;
    border: 0 solid transparent;
    border-radius: 0;
    cursor: pointer;
    display: inline-block;
    padding: 5px;
    background-color: transparent;
}

.inner {
    margin: -1px 0 0;
    position: relative;
    width: 100%
}

.hamburger .bar {
    background-color: var(--white);
    border-radius: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    width: 100%
}

.hamburger .bar:first-child {
    top: -10px;
}

.hamburger .bar:nth-child(3) {
    top: 10px;
}

.hamburger.active .bar:first-child {
    transform: rotate(45deg) translate3d(7px, 7px, 0);
    transition-delay: .1s
}

.hamburger.active .bar:nth-child(2) {
    transform: scaleX(0);
    transition-delay: .05s
}

.hamburger.active .bar:nth-child(3) {
    transform: rotate(-45deg) translate3d(7px, -7px, 0);
    transition-delay: .1s
}

.page {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: large;
}

.pageContainer {
    position: relative;
    min-height: calc(100vh - 4.5rem);
}

.response {
    height: 25px;
    width: 25px;
    padding: 0 2rem;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
}

.success {
    background-image: url('/assets/svg/success.svg');
}

.failure {
    background-image: url('/assets/svg/failure.svg');
}

.submit {
    background-image: url('/assets/svg/submit.svg');
}

.magnifier {
    background-image: url('/assets/svg/magnifier.svg');
}

.resetBtn,
.submitBtn {
    width: 150px;
}

.linkItem {
    display: inline-flex;
    flex-direction: row;
    align-items: center;
    color: var(--blue);
    margin: 0;
}

.white {
    color: var(--white) !important;
}

.linkItem:hover {
    cursor: pointer;
    text-decoration: underline;
}

.decoration {
    padding: 0 .2rem;
}

@media(min-width:768px) {
    .pageContainer {
        min-height: calc(100vh - 7rem);
    }
}