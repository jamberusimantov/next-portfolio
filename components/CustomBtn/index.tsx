import React, { forwardRef } from 'react';
import type { MouseEventHandler, CSSProperties } from 'react';
import styles from './CustomBtn.module.css'
import { JsxElement } from 'typescript';


interface iCustomBtnProps {
    id?: string,
    className?: string,
    text?: string,
    status?: string,
    handleClick?: MouseEventHandler<HTMLButtonElement>;
    handleMouseEnter?: MouseEventHandler<HTMLButtonElement>;
    handleMouseLeave?: MouseEventHandler<HTMLButtonElement>;
    type?: keyof { "button": "", "submit": "", "reset": "" };
    title?: string;
    style?: CSSProperties;
    children?: JsxElement
}

const CustomBtn = forwardRef<HTMLButtonElement, iCustomBtnProps>((props, ref) => {
    const { id, className, text, status, handleClick, type = "button", handleMouseEnter, handleMouseLeave } = props;
    return <button
        ref={ref}
        id={id}
        className={`${className ?? ""} ${styles.button}`}
        onClick={handleClick}
        onMouseEnter={handleMouseEnter}
        onMouseLeave={handleMouseLeave}
        type={type}
        children={[
            text && <p key={0}>{text}</p>,
            status && <div className={status} key={1}></div>
        ]}
    />

});

export default CustomBtn;