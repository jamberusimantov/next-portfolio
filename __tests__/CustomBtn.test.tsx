import React from "react";
import { describe, expect, test } from '@jest/globals';
import { create } from 'react-test-renderer';
import CustomBtn from '../components/CustomBtn';


describe('CustomBtn', () => {
    test('renders correctly', () => {
        let tree = create(<CustomBtn text="JAVA" status="response submit" />).toJSON();
        if (!tree) return;
        tree = tree instanceof Array ? tree[0] : tree;
        expect(tree).toMatchSnapshot();
    })
})