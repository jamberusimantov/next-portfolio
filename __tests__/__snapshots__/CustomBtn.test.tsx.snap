// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomBtn renders correctly 1`] = `
<button
  className=" button"
  type="button"
>
  <p>
    JAVA
  </p>
  <div
    className="response submit"
  />
</button>
`;
